EIPS/eip-7873.md

```diff
@@ -156,19 +156,20 @@ At the start of the block in which [EIP-3540](./eip-3540.md) activates, set the
     let size := calldatasize()
     if lt(size, 64) { revert(0, 0) }
 
-    let tx_initcode_hash := calldataload(0)
-    let salt := calldataload(32)
-
-    mstore8(0, 0xff)  // a magic value to ensure a specific preimage space
-    calldatacopy(1, 0, size)  // copy tx_initcode_hash, salt and init_data to memory to hash
-    let commitment_size := add(size, 1)
-    let final_salt := keccak256(0, commitment_size)
+    calldatacopy(0, 0, size)  // copy tx_initcode_hash, salt and init_data to memory to hash
+    let final_salt := keccak256(0, size)
 
+    let tx_initcode_hash := calldataload(0)
```

Comment:
This could also be `mload(0)`. Is there any reason to prefer one over the other? Are there any planned gas schedule changes that would impact here?

Reply:
`calldataload(0)` is simpler/safer, as it reads from an immutable buffer, independent from code that wrote that data into memory.

Not sure about gas schedule changes, but anyway I'd prefer simplicity today over performance in the future.